src/common/docker-compose/index.ts

```diff
@@ -351,6 +351,17 @@ export class DockerComposeUtils {
         }
       }
 
+      if (service.build) {
+        // Add labels to enrich image filtering
+        if (!service.build.labels) {
+          service.build.labels = [];
+        }
+        service.build.labels.push(DOCKER_IMAGE_LABEL);
+
+        const { component_name } = ResourceSlugUtils.parse(node.config.metadata.ref);
+        service.build.labels.push(`component=${component_name}`);
```

Comment:
Is the label `component` specific enough? People can add their own labels and this could easily conflict

Reply:
fair point, can prefix with `architect` (not sure if `.` is allowed but maybe `architect.component`) just to namespace it/free up more general words for people